fili-core/src/test/groovy/com/yahoo/bard/webservice/druid/serializers/DimensionToDefaultDimensionSpecSpec.groovy

```diff
@@ -0,0 +1,96 @@
+ */
+class DimensionToDefaultDimensionSpecSpec extends Specification {
+
+    static ObjectMapper objectMapper
+    static QueryBuildingTestingResources resources
+    static Set<Interval> intervals
+    static DruidQueryBuilder builder
+    static DefaultPhysicalTableResolver resolver
+
+    def initDefault(DataApiRequest apiRequest, Dimension dim) {
+        intervals = [new Interval(new DateTime("2015"), Hours.ONE)]
+        LogicalMetric lm1 = new LogicalMetric(resources.simpleTemplateQuery, new NoOpResultSetMapper(), "lm1", null)
+
+        apiRequest.getTable() >> resources.lt12
+        apiRequest.getGranularity() >> HOUR.buildZonedTimeGrain(UTC)
+        apiRequest.getTimeZone() >> UTC
+        apiRequest.getDimensions() >> ([dim] as Set)
+        apiRequest.getLogicalMetrics() >> ([lm1] as Set)
+        apiRequest.getIntervals() >> intervals
+        apiRequest.getFilterDimensions() >> []
+        apiRequest.getTopN() >> OptionalInt.empty()
+        apiRequest.getSorts() >> ([] as Set)
+        apiRequest.getCount() >> OptionalInt.empty()
+    }
+
+    def setupSpec() {
+        objectMapper = new ObjectMapper()
+        resources = new QueryBuildingTestingResources()
+        resolver = new DefaultPhysicalTableResolver(new PartialDataHandler(), new DefaultingVolatileIntervalsService())
+        builder = new DruidQueryBuilder(resources.logicalDictionary, resolver)
+    }
+
+    def "Testing serialization with equal apiName and physicalName, serialize to apiName string"() {
+        setup:
+        DataApiRequest apiRequest = Mock(DataApiRequest)
+        initDefault(apiRequest, resources.d1)
+
+        when:
+        DruidAggregationQuery<?> dq = builder.buildQuery(apiRequest, resources.simpleTemplateQuery)
+
+        then:
+        objectMapper.writeValueAsString(dq).contains('"dimensions":["dim1"]')
+    }
+
+    def "Testing serialization with different apiName and physicalName, serialize to default dimension spec"() {
+        setup:
+        DataApiRequest apiRequest = Mock(DataApiRequest)
+        initDefault(apiRequest, resources.d3)
+
+        when:
+        DruidAggregationQuery<?> dq = builder.buildQuery(apiRequest, resources.simpleTemplateQuery)
+
+        then:
+        objectMapper.writeValueAsString(dq).contains('"dimensions":[{"type":"default","dimension":"age_bracket","outputName":"ageBracket"}]')
+    }
+
+    def "Testing serialization with nested query as datasource, only serializing the inner-most query dimension"() {
+        setup:
+        DataApiRequest apiRequest = Mock(DataApiRequest)
+        initDefault(apiRequest, resources.d3)
+
+        when:
+        DruidAggregationQuery<?> dq = builder.buildQuery(apiRequest, resources.simpleNestedTemplateQuery)
+
+        then:
+        objectMapper.writeValueAsString(dq).matches("(.*)\"dimensions\":\\[\\{\"type\":\"default\",\"dimension\":\"age_bracket\",\"outputName\":\"ageBracket\"\\}\\](.*)\"dimensions\":\\[\"ageBracket\"\\](.*)")
```

Comment:
Rather than using a complex regex, we should just use 2 different `contains` calls, one for each of the things we're looking for.